app/fighters/knight.py

```diff
@@ -0,0 +1,28 @@
+class Knight:
+    def __init__(self, knight_stat: dict) -> None:
+        self.name = knight_stat["name"]
+        self.hp = knight_stat["hp"]
+        self.power = knight_stat["power"] + knight_stat["weapon"]["power"]
+        self.protection = self.count_protection(knight_stat)
+        self.use_potion(knight_stat)
+
+    @staticmethod
+    def count_protection(knight_stat: dict) -> int:

```

Comment:
You've implemented the count_protection method well, but it might be worth making it private (add a _ before it) to indicate that it's for internal use. Anyway, everything looks great, good job!